### PR TITLE
chore: Performance reporter changes

### LINKED ISSUE #19403: Unification: Performance Testing
### What would you like?

Set up baseline tests for performance to ensure there are no regressions on unification release. 

### Why is this needed?

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by BlueWinds**
- Closes #19403 

### User facing changelog
No user facing changes.

### Additional details
Updates the performance-reporter to send honeycomb events for specs as well as individual tests, bundling this all together into honeycomb traces for easier reporting / analysis.

For an example of the new tracing, see https://ui.honeycomb.io/cypress/datasets/systemtest-performance/result/6g9856meLqn/trace/aTmoUYdQLEb?span=22f3e7f4-3153-4f55-b758-a285dc4ccf10.

### PR Tasks
- [N/A] Have tests been added/updated?
- [N/A] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [N/A] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [N/A] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [N/A] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: system-tests/lib/performance-reporter.js
Comment (BlueWinds) [inline system-tests/lib/performance-reporter.js:57]: This log left in intentionally, so it shows up in circleci and gives us a way (not an easy way, but at least *a* way) to see the traceid associated with a specific build.

File: system-tests/lib/performance-reporter.js
Comment (flotwig) [inline system-tests/lib/performance-reporter.js:55]: nit: IMO it would be easier for new people coming in to debug this if this "script" was in `system-tests/scripts` and not just conditionally running if this `module` is `main`. Separation of the script and the lib is how most of our scripts are set up.

**Review by flotwig — DISMISSED** [review]
:cool: :exclamation:

File: system-tests/lib/performance-reporter.js
Comment (BlueWinds) [reply system-tests/lib/performance-reporter.js]: Done.

**Review by flotwig — APPROVED** [review]

**Review by tgriesser — CHANGES_REQUESTED** [review]
Very cool - I think the main thing that's missing, at least from the trace you linked, is a "name" identifier of each span... I think in this case it'd be `ci_run`, `spec_execution`, `test_execution`

Also, do you think there's a way we can get a `durationMs` added to the root span?

**Comment by BlueWinds** [comment]
> Very cool - I think the main thing that's missing, at least from the trace you linked, is a "name" identifier of each span... I think in this case it'd be `ci_run`, `spec_execution`, `test_execution`

Adding.

> Also, do you think there's a way we can get a `durationMs` added to the root span?

I suppose it would be possible - I could add a new circleci step which depends on all the system-test steps, and delay sending the root span until all the others are executed. It would need a way to pass the 'start' timestamp between circleci stages (probably by writing it to a file in the system-tests node_modules cache).

So "yes, but not trivially" - probably as a separate PR at a later date.

**Review by flotwig — APPROVED** [review]

**Comment by BlueWinds** [comment]
Since Tim is out the rest of the week and I addressed his requested changes, went ahead and merged without waiting for re-approval.

**Pull request merged**
